internal/xds/xdsclient/xdsresource/unmarshal_rds_test.go

```diff
@@ -722,6 +724,95 @@ func (s) TestRDSGenerateRDSUpdateFromRouteConfiguration(t *testing.T) {
 	}
 }
 
+func (s) TestGenerateRDSUpdateFromRouteConfigurationWithAutoHostRewrite(t *testing.T) {
+	const (
+		clusterName = "clusterName"
+		ldsTarget   = "lds.target.good:1111"
+	)
+
+	buildRouteConfig := func() *v3routepb.RouteConfiguration {
+		return &v3routepb.RouteConfiguration{
+			Name: "routeName",
+			VirtualHosts: []*v3routepb.VirtualHost{{
+				Domains: []string{ldsTarget},
+				Routes: []*v3routepb.Route{{
+					Match: &v3routepb.RouteMatch{PathSpecifier: &v3routepb.RouteMatch_Prefix{Prefix: "/"}},
+					Action: &v3routepb.Route_Route{
+						Route: &v3routepb.RouteAction{
+							ClusterSpecifier:     &v3routepb.RouteAction_Cluster{Cluster: clusterName},
+							HostRewriteSpecifier: &v3routepb.RouteAction_AutoHostRewrite{AutoHostRewrite: &wrapperspb.BoolValue{Value: true}},
```

Comment:
We probably also need test cases for when the `AutoHostRewrite` field in the proto is `false`.